components/error_page/error_title.jsx

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {ErrorPageTypes} from 'utils/constants.jsx';
+import * as Utils from 'utils/utils.jsx';
+
+export default function ErrorTitle({type, title}) {
+    let errorTitle = null;
+    if (type) {
+        switch (type) {
+        case ErrorPageTypes.LOCAL_STORAGE:
+            errorTitle = (
+                <FormattedMessage
+                    id='error.local_storage.title'
+                    defaultMessage='Cannot Load Mattermost'
+                />
+            );
+            break;
+        case ErrorPageTypes.PERMALINK_NOT_FOUND:
+            errorTitle = (
+                <FormattedMessage
+                    id='permalink.error.title'
+                    defaultMessage='Message Not Found'
+                />
+            );
+            break;
+        case ErrorPageTypes.OAUTH_MISSING_CODE:
+            errorTitle = (
+                <FormattedMessage
+                    id='error.oauth_missing_code.title'
+                    defaultMessage='Mattermost needs your help'
+                />
+            );
```

Comment:
This case fixes the ticket; others are just code refactoring and additional unit tests.